scripts/play.ts

```diff
@@ -0,0 +1,550 @@
+  start: boolean;
+  new: boolean;
+  framework?: "next" | "nuxt" | "sveltekit";
+  stage: "staging" | "prod" | "development";
+  help: boolean;
+};
+
+type DryRunOption = {
+  /**
+   * If `true`, commands are not executed.
+   */
+  dryRun?: Args["dry-run"];
+};
+
+/**
+ * The root command.
+ */
+async function run(): Promise<void> {
+  process.on("uncaughtException", handleUncaughtException);
+
+  // Ensure dependencies are isolated from the monorepo.
+  if (!(await pathExists(new URL("./yarn.lock", PLAYGROUNDS_ROOT)))) {
+    await fs.writeFile(new URL("./yarn.lock", PLAYGROUNDS_ROOT), "");
+  }
+
+  const args = mri<Args>(process.argv.slice(2), {
+    boolean: ["help", "dry-run", "start", "new"],
+    string: ["framework", "stage"],
+    alias: { h: "help", n: "dry-run", f: "framework" },
+    default: {
+      "dry-run": false,
+      start: true,
+      new: false,
+      stage: DEFAULT_STAGE,
+    },
+  });
+
+  if (args.help) {
+    console.info(
+      `
+Usage:
+    yarn play [options...] [name]
+
+Options:
+    --new            Create a new playground
+    --framework, -f  Specify the playground's framework (next, nuxt, sveltekit) (default: next)
+    --stage          Specify the playground's stage (staging, prod, development) (default: staging)
+    --no-start       Do not start Slice Machine and the website
+    --dry-run, -n    Show what would have happened
+    --help, -h       Show help text
+
+Arguments:
+    [name]           The name of the playground to create or start
+`.trim(),
+    );
+
+    return;
+  }
+
+  if (!["prod", "staging", "development"].includes(args.stage)) {
+    throw new CommandError(`Unsupported stage: ${args.stage}`);
+  }
+
+  if (
+    args.framework &&
+    !["next", "nuxt", "sveltekit"].includes(args.framework)
+  ) {
+    throw new CommandError(`Unsupported framework: ${args.framework}`);
+  }
+
+  let [playgroundName] = args._;
+  const didProvidePlaygroundName = Boolean(playgroundName);
+
+  if (!playgroundName) {
+    if (args.new) {
+      playgroundName = createRandomName();
+    } else {
+      playgroundName =
+        (await getLastModifiedEntry(PLAYGROUNDS_ROOT, {
+          dirOnly: true,
+          exclude: [".yarn", "node_modules"],
+        })) ?? createRandomName();
+    }
+  }
+
+  if (!playgroundName.startsWith("play-")) {
+    playgroundName = `play-${playgroundName}`;
+  }
+
+  const playgroundDir = new URL(`./${playgroundName}/`, PLAYGROUNDS_ROOT);
+  const playgroundExists = await pathExists(playgroundDir);
+
+  if (playgroundExists) {
+    if (args.new) {
+      throw new CommandError(
+        `A playground named ${playgroundName} already exists. Exiting.`,
+      );
+    }
+
+    if (didProvidePlaygroundName) {
+      console.info(`Using playground: ${chalk.green(playgroundName)}`);
+    } else {
+      console.info(
+        `Reusing most recent playground: ${chalk.green(playgroundName)}`,
+      );
+    }
+
+    console.info('  (Use "--new" to create a new playground)');
```

Comment:
This line should only be reached if `--new` is false since we throw above if `--new` is true: https://github.com/prismicio/slice-machine/blob/bfb737b97ecaa25ad0f3797b5416a2c39f9036f4/scripts/play.ts#L117-L121